pkg/config/decode_test.go

```diff
@@ -0,0 +1,60 @@
+package config
+
+import (
+	"testing"
+
+	"github.com/pelletier/go-toml"
+	"github.com/stretchr/testify/require"
+)
+
+func TestDecode(t *testing.T) {
+	treeMap := map[string]interface{}{
+		"core": map[string]interface{}{
+			"dcos_url":   "https://dcos.example.com",
+			"ssl_verify": "false",
+			"timeout":    15,
+			"reporting":  true,
+		},
+		"cluster": map[string]interface{}{
+			"name": "mr-cluster",
+		},
+	}
+	tree, _ := toml.TreeFromMap(treeMap)
+	conf, err := decode(tree, nil)
+	require.NoError(t, err)
+
+	require.Equal(t, "https://dcos.example.com", conf.Core.URL)
+	require.Equal(t, true, conf.Core.TLS.Insecure())
+	require.Equal(t, 15, conf.Core.Timeout)
+	require.Equal(t, true, conf.Core.Reporting)
+	require.Equal(t, "mr-cluster", conf.Cluster.Name)
+}
+
+func TestDecodeWithEnvVars(t *testing.T) {
+	// core.dcos_url should get overriden by DCOS_URL.
+	// cluster.name shouldn't get overriden by DCOS_CLUSTER_NAME.
```

Comment:
According to the Go codebase, it's OK to have contractions in comments :+1: https://github.com/golang/go/blob/29461ccc7f7b6a73bc0104731cfbe0c2cbe86733/src/go/parser/interface.go#L131